airbyte-webapp/src/pages/SpeakeasyRedirectPage/SpeakeasyRedirectPage.tsx

```diff
@@ -0,0 +1,67 @@
+import { Suspense } from "react";
+import React from "react";
+import { FormattedMessage } from "react-intl";
+import { Navigate, useNavigate } from "react-router-dom";
+
+import { LoadingPage } from "components/LoadingPage";
+
+import { useAppMonitoringService } from "hooks/services/AppMonitoringService";
+import { useSpeakeasyRedirect } from "packages/cloud/services/cloudApi/useSpeakeasyRedirect";
+import { RoutePaths } from "pages/routePaths";
+import { ErrorOccurredView } from "views/common/ErrorOccurredView";
+
+export const SpeakeasyRedirectPage = () => {
+  const { trackError } = useAppMonitoringService();
+
+  return (
+    <SpeakeasyErrorBoundary trackError={trackError}>
+      <SpeakeasyLoginRedirect />
+    </SpeakeasyErrorBoundary>
+  );
+};
+
+const SpeakeasyLoginRedirect = () => {
+  const { redirectUrl } = useSpeakeasyRedirect();
+
+  return (
+    <Suspense fallback={<LoadingPage />}>
+      {redirectUrl ? <Navigate to={redirectUrl} /> : <CloudApiErrorView />}
+    </Suspense>
+  );
+};
+
+const CloudApiErrorView = () => {
+  const navigate = useNavigate();
+  return (
+    <ErrorOccurredView
+      message={<FormattedMessage id="cloudApi.loginCallbackUrlError" />}
+      ctaButtonText={<FormattedMessage id="ui.goBack" />}
+      onCtaButtonClick={() => {
+        navigate(RoutePaths.Root);
+      }}
+    />
+  );
+};
+
+interface SpeakeasyErrorBoundaryProps {
+  trackError: (e: Error, context?: Record<string, unknown>) => void;
```

Comment:
Too bad `AppMonitoringService.tsx` doesn't export a function type for each `AppMonitoringServiceProviderValue` field; it seems both inconvenient and brittle to have to manually-redefine this signature at the usage sites (and of course the rules of hooks prevent us from having access to `typeof trackError` in top-level module scope).

Reply:
Yeah this is true, however just to clarify, we won't need to manually redefine the function signature at all usage sites. This is a pretty specific scenario where we can't use `useAppMonitoringService()` directly in the component since it's an error boundary and must be a class component. Everywhere else we can just do `const { trackAction, trackError } = useAppMonitoringService()` and get the types. In any case, it doesn't hurt to export the function types for this specific component!